File: front_react/src/domain/LearningSettings.ts
import type {Lang} from './Lang.ts';
import type {Identified} from './identity/Identified.ts';
import type {TagSmall} from './TagSmall.ts';

export type LearningSettings = {
  readonly learnLangs: ReadonlyArray<Lang>,
  readonly knownLangs: ReadonlyArray<Lang>,
  readonly tags: ReadonlyArray<Identified<TagSmall>>
}

export const LearningSettings = {
  empty: {
    learnLangs: [],
    knownLangs: [],
    tags: []
  } as LearningSettings
}

File: services/modules/domain/src/main/scala/solutions/s4y/vocabla/domain/LearningSettings.scala
package solutions.s4y.vocabla.domain

import solutions.s4y.vocabla.domain.identity.{
  Identified,
  Identifier,
  IdentifierSchema
}
import zio.Chunk
import zio.json.{DeriveJsonCodec, JsonCodec}

final case class LearningSettings(
    learnLanguages: Chunk[Lang.Code],
    knownLanguages: Chunk[Lang.Code],
    tags: Chunk[Identified[TagSmall]]
)

object LearningSettings:
  val emptyLearningSettings: LearningSettings = LearningSettings(
    learnLanguages = Chunk.empty,
    knownLanguages = Chunk.empty,
    tags = Chunk.empty
  )

  given (using IdentifierSchema): JsonCodec[LearningSettings] =
    DeriveJsonCodec.gen[LearningSettings]
